internal/services/mongocluster/mongo_cluster_resource.go

```diff
@@ -0,0 +1,572 @@
+			if metadata.ResourceData.HasChange("administrator_password") {
+				model.Properties.Administrator = &mongoclusters.AdministratorProperties{
+					UserName: pointer.To(state.AdministratorUserName),
+					Password: pointer.To(state.AdministratorPassword),
+				}
+			}
+
+			if metadata.ResourceData.HasChange("high_availability_mode") {
+				model.Properties.HighAvailability = &mongoclusters.HighAvailabilityProperties{
+					TargetMode: pointer.To(mongoclusters.HighAvailabilityMode(state.HighAvailabilityMode)),
+				}
+			}
+
+			if metadata.ResourceData.HasChange("public_network_access_enabled") {
+				if state.PublicNetworkAccessEnabled {
+					model.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessEnabled)
+				} else {
+					model.Properties.PublicNetworkAccess = pointer.To(mongoclusters.PublicNetworkAccessDisabled)
+				}
+			}
+
+			if metadata.ResourceData.HasChange("storage_size_in_gb") {
+				model.Properties.Storage = &mongoclusters.StorageProperties{
+					SizeGb: pointer.To(state.StorageSizeInGb),
+				}
+			}
+
+			if metadata.ResourceData.HasChange("tags") {
+				model.Tags = pointer.To(state.Tags)
+			}
+
+			if metadata.ResourceData.HasChange("version") {
+				model.Properties.ServerVersion = pointer.To(state.Version)
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, *id, *model); err != nil {
+				return fmt.Errorf("updating %s: %+v", *id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (r MongoClusterResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.MongoCluster.MongoClustersClient
+
+			id, err := mongoclusters.ParseMongoClusterID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(*id)
+				}
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := MongoClusterResourceModel{
+				Name:              id.MongoClusterName,
+				ResourceGroupName: id.ResourceGroupName,
+			}
+
+			if model := resp.Model; model != nil {
+				state.Location = location.NormalizeNilable(&model.Location)
+
+				if props := model.Properties; props != nil {
+					// API doesn't return the value of administrator_password
+					state.AdministratorPassword = metadata.ResourceData.Get("administrator_password").(string)
+
+					// API doesn't return the value of create_mode
+					state.CreateMode = metadata.ResourceData.Get("create_mode").(string)
+
+					if v := props.Administrator; v != nil {
+						state.AdministratorUserName = pointer.From(v.UserName)
+					}
+
+					if v := props.Replica; v != nil {
+						// API doesn't return the value of source_location
+						state.SourceLocation = metadata.ResourceData.Get("source_location").(string)
```

Comment:
Same here

Reply:
The response from the service team is that this is by design. Please refer to: https://github.com/Azure/azure-rest-api-specs/issues/31266